src/Hack.sol

```diff
@@ -843,3 +843,66 @@ contract Ticketing {
         tickets[msg.sender]++;
     }
 }
+
+/* Exercise 18 */
+
+/// @dev This contract represents a Battle Royale game where participants compete to become the "king" by achieving the lowest score.
+///      Participants can challenge the current king by deploying their own challenger contract.
+///      The challenger must return a non-empty response when called.
+///      If the challenger's address is lower than the current king's challenger address, they dethrone the previous king and become the new king.
+///      The rewards are distributed among the participants based on the time they held the king position.
+contract BattleRoyale {
+    uint256 public constant _duration = 1 weeks;
+    uint256 public constant _totalReward = 10 ether;
```

Comment:
we put constant in capital letters though. 

And it's the internal, private state variables that we put the underscore sorry for not being clear enough 😆